components/health_controller/src/reporters.rs

```diff
@@ -0,0 +1,234 @@
+// Copyright 2024 TiKV Project Authors. Licensed under Apache-2.0.
+
+use std::{
+    sync::Arc,
+    time::{Duration, Instant},
+};
+
+use kvproto::pdpb;
+use pdpb::SlowTrend as SlowTrendPb;
+use prometheus::IntGauge;
+
+use crate::{
+    slow_score::{SlowScore, SlowScoreTickResult},
+    trend::{RequestPerSecRecorder, Trend},
+    HealthController, HealthControllerInner, RaftstoreDuration,
+};
+
+pub struct RaftstoreReporterConfig {
+    pub inspect_interval: Duration,
+
+    pub unsensitive_cause: f64,
+    pub unsensitive_result: f64,
+    pub net_io_factor: f64,
+
+    pub cause_spike_filter_value_gauge: IntGauge,
+    pub cause_spike_filter_count_gauge: IntGauge,
+    pub cause_l1_gap_gauges: IntGauge,
+    pub cause_l2_gap_gauges: IntGauge,
+
+    pub result_spike_filter_value_gauge: IntGauge,
+    pub result_spike_filter_count_gauge: IntGauge,
+    pub result_l1_gap_gauges: IntGauge,
+    pub result_l2_gap_gauges: IntGauge,
+}
+
+pub struct RaftstoreReporter {
+    health_controller_inner: Arc<HealthControllerInner>,
+    slow_score: SlowScore,
+    slow_trend: SlowTrendStatistics,
+    is_healthy: bool,
+}
+
+impl RaftstoreReporter {
```

Comment:
If the read pool information need to be collected, the pattern is:
- Adding a `ReadPoolReporter` reporter holding a `Arc` to the `HealthController`
- Adding related fields inside the  `HealthController`

right?

Besides, a similar question  like above, why holding an `inner` object?

Reply:
1. Yes, but it's not an `Arc` to `HealthController`. Instead, it should be a `Arc<HealthControllerInner>` or a clone of `HealthController`.
2. Separating an "inner" is just to hide the `Arc` inside the `HealthController` and avoid explicitly writing `Arc` when using it, which I think is a common pattern.